include/alpaka/atomic/AtomicUniformCudaHipBuiltIn.hpp

```diff
@@ -185,14 +185,70 @@ namespace alpaka
                 if constexpr(std::numeric_limits<unsigned int>::max() == std::numeric_limits<unsigned long>::max())
                     return ::atomicSub(reinterpret_cast<unsigned int*>(addr), static_cast<unsigned int>(value));
                 else // LP64
-                    static_assert(
-                        !sizeof(THierarchy),
-                        "atomicOp<AtomicSub, AtomicUniformCudaHipBuiltIn, unsigned long int> is only supported when "
-                        "sizeof(unsigned long int) == 4");
+                {
+                    return ::atomicAdd(
+                        reinterpret_cast<unsigned long long int*>(addr),
+                        static_cast<unsigned long long int>(-value));
```

Comment:
```suggestion
                    return ::atomicAdd(addr, -value);
```
Why not stay with the same datatype when forwarding to `atomicAdd`?

Reply:
I think I simply emulated the same behaviour as of atomic add, recasting depending on LLP64 or not.

Reply:
Ok, but if `atomicAdd` already does that for you, then just delegate to it :)